### PR TITLE
Fix edit issues

### LINKED ISSUE #189: [PE-D][Tester C] Error when editting module
If I pass in extremely large number to edit the credit to (edit mod cs2040s -cr 9999999...), there is NumberFormatException (which should be caught, but didn't get caught). This lead to UI errors.

-------------
Labels: `type.FunctionalityBug` `severity.High`
original: ngquyduc/ped#3

### PR DESCRIPTION
**Pull request by HoJunHao2000**
* Fix NPEs in edit module command
* Modified module credit to be between 1-2 digits long only
* Modified module code to be between 6-9 characters long only

Not sure which "too long" UI issues it might resolve or even if this modification is allowed. 

Modified error message to inform about the input limitation

Resolves #189

### CODE REVIEW

File: src/main/java/modtrekt/model/module/ModCode.java [hunk 1]
```diff
@@ -9,8 +9,9 @@
  */
 public class ModCode implements Comparable<ModCode> {
 
-    public static final String MESSAGE_CONSTRAINTS = "Code should contain alphanumeric characters and no white space";
-    private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
+    public static final String MESSAGE_CONSTRAINTS =
+            "Code should contain alphanumeric characters, no white space, and 6 to 9 characters long";
```
Comment (vvidday) [inline src/main/java/modtrekt/model/module/ModCode.java]: Can we make this more specific, e.g.
`Code should only contain alphanumeric characters, should not contain white space and should be between 6 and 9 characters long`?

**Review by vvidday — CHANGES_REQUESTED** [review]
LGTM thanks for the catch on the NPE!

Just one minor phrasing suggestion for the error message, + can you also update the UG with the new restrictions? Thanks!

File: src/main/java/modtrekt/model/module/ModCode.java (hunk 1 above)
Comment (HoJunHao2000) [reply src/main/java/modtrekt/model/module/ModCode.java]: done. the UG changes ill do it in the other PR #210

**Review by vvidday — APPROVED** [review]
Thanks, LGTM!

**Pull request merged**